### ISSUE TITLE
Bray-Curtis: `Data must be symmetric and cannot contain NaNs.`

### ISSUE THREAD

**Issue by sjanssen2** [qiime2/q2-diversity#262]
I encountered the error message
```
Plugin error from diversity:

  Data must be symmetric and cannot contain NaNs.

Debug info has been saved to /tmp/qiime2-q2cli-err-4o_gtofm.log
```
when I tried to compute Bray-Curtis beta diversity distances for the attached [test.txt](https://github.com/qiime2/q2-diversity/files/3578483/test.txt) feature table (is lists cytokine concentrations not sequence reads, but that is not the cause of the problem):

 - `biom convert --to-hdf5 -i test.txt -o input.biom`
 - `qiime tools import --input-path input.biom --type FeatureTable[Frequency] --output-path input.qza`
 - `qiime diversity beta --i-table input.qza --p-metric braycurtis --p-n-jobs 1 --o-distance-matrix braycurtis`

It works for other metrics, e.g. Jaccard

`qiime diversity beta --i-table input.qza --p-metric jaccard --p-n-jobs 1 --o-distance-matrix jaccard`

I figure, the problem is caused by samples 65, 66, 67, 68 and 69 which are `0` for all features. However, we should return a more speaking message that points to the input feature table and NOT to the resulting distance matrix, which could not even been computed. It took me some time to figure this out.

**Comment by ebolyen** [comment]
Yeah a better error message is a really good idea. I believe @ChrisKeefe has some related work for alpha diversity (NaN vs filter) which can be similarly applied for beta diversity (which he will also be tackling soon). This is all happening in the context of [q2-diversity-lib](https://github.com/qiime2/q2-diversity-lib)

**Comment by cherman2** [comment]
I just tested this and this error still exists.
